src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/Common/DbConnectionOptions.cs

```diff
@@ -16,40 +16,25 @@ internal partial class DbConnectionOptions
         // used by pooling classes so it is much easier to verify correctness
         // when not worried about the class being modified during execution
 
-        public DbConnectionOptions(string connectionString, Dictionary<string, string> synonyms)
+        public DbConnectionOptions(string connectionString)
         {
-            _parsetable = new Dictionary<string, string>();
             _usersConnectionString = ((null != connectionString) ? connectionString : "");
-
-            // first pass on parsing, initial syntax check
-            if (0 < _usersConnectionString.Length)
-            {
-                _keyChain = ParseInternal(_parsetable, _usersConnectionString, true, synonyms, false);
-                HasPasswordKeyword = (_parsetable.ContainsKey(KEY.Password) || _parsetable.ContainsKey(SYNONYM.Pwd));
-                HasUserIdKeyword = (_parsetable.ContainsKey(KEY.User_ID) || _parsetable.ContainsKey(SYNONYM.UID));
-            }
         }
 
         protected DbConnectionOptions(DbConnectionOptions connectionOptions)
-        { // Clone used by SqlConnectionString
+        { 
+            // Clone used by SqlConnectionString
             _usersConnectionString = connectionOptions._usersConnectionString;            
-            _parsetable = connectionOptions._parsetable;
-            _keyChain = connectionOptions._keyChain;
-            HasPasswordKeyword = connectionOptions.HasPasswordKeyword;
-            HasUserIdKeyword = connectionOptions.HasUserIdKeyword;
         }
 
-        public bool IsEmpty => _keyChain == null;
-
-        internal bool TryGetParsetableValue(string key, out string value) => _parsetable.TryGetValue(key, out value);
+        protected static bool TryGetParsetableValue(Dictionary<string, string> parsetable, string key, out string value) => parsetable.TryGetValue(key, out value);
```

Comment:
It seems you've made some methods `static` as you're passing `parsetable` to these methods and now they are act as utility. Could you also update other similar APIs too and make them uniform?

Reply:
If I run ReSharper over the project it suggest making 156 more static method.
Would you like to make them in a separate PR ?

Reply:
If they are related to this PR, you can modify them here. Otherwise, classifying them on separate perf PR makes it easy to review.